pkg/registry/apis/query/testdata/legacy-ds-uid.json

```diff
@@ -0,0 +1,82 @@
+{
+  "description": "datasource uid is variable",
+  "input": {
+    "from": "now-6",
+    "to": "now",
+    "queries": [
+      {
+        "refId": "C",
+        "queryType": "",
+        "datasourceUid": "__expr__",
+        "type": "reduce",
+        "reducer": "last",
+        "datasourceId": 100,
+        "datasource": {
+          "type": "__expr__",
+          "uid": "__expr__"
+        },
+        "expression": "$B"
+      },
+      {
+        "refId": "A",
+        "queryType": "",
+        "datasourceUid": "XYX",
+        "datasource": {
+          "uid": "${datasource}"
+        }
+      },
+      {
+        "refId": "B",
+        "queryType": "",
+        "datasourceUid": "__expr__",
+        "type": "math",
+        "datasource": {
+          "type": "__expr__",
+          "uid": "__expr__"
+        },
+        "expression": "$A + 10"
+      }
+    ]
+  },
+  "expect": {
+    "requests": [
+      {
+        "uid": "XYX",
+        "pluginId": "plugin-bbb",
+        "request": {
+          "from": "now-6",
+          "to": "now",
+          "queries": [
+            {
+              "refId": "A",
+              "datasource": {
+                "uid": "${datasource}",
+                "type": ""
+              },
+              "datasourceUid": "XYX"
+            }
+          ]
+        }
+      }
+    ],
+    "expressions": [
+      {
+        "id": 2,
+        "refId": "B",
+        "type": "math",
+        "properties": {
+          "expression": "$A + 10"
+        }
+      },
```

Comment:
I like test cases to be as small and simple as possible. To me it seems this block is outside the scope of the new code we're adding here, and the new specification we're trying to set for this code. Can we remove this block completely? Can we shorten this test in any other ways?

Reply:
Ah yeah, good call. I was _very_ focused on this case working with expressions that I forgot to clear out the test :trophy: